pandas/_libs/groupby.pyx

```diff
@@ -433,5 +436,212 @@ def _group_add(floating[:, :] out,
 group_add_float32 = _group_add['float']
 group_add_float64 = _group_add['double']
 
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _group_prod(floating[:, :] out,
+                int64_t[:] counts,
+                floating[:, :] values,
+                const int64_t[:] labels,
+                Py_ssize_t min_count=0):
+    """
+    Only aggregates on axis=0
+    """
+    cdef:
+        Py_ssize_t i, j, N, K, lab, ncounts = len(counts)
+        floating val, count
+        ndarray[floating, ndim=2] prodx, nobs
+
+    if not len(values) == len(labels):
```

Comment:
!=

Reply:
I made the minimum amount of changes when I refactored from tempita to fused types.

So just to make sure, you would like me to take this opportunity to the fix small issues inside the functions themselves?

Reply:
Yes please.  Be sure to comment to point out where you have made changes.

Reply:
generally when moving large blocks of code best to just move them and not make any other changes. generally prefer to just have these changes in a followup; if very small ok though.

Reply:
Fixed.